### DISCUSSION TITLE
Tokens / cryptocurrency on top of Locutus

### DISCUSSION THREAD

**Discussion by sfsfrgrg** [freenet/freenet-core#102] [General]
Is this software free? Is it a blockchain? I don't like blockchain 
 I don't like tokens

**Comment by sfsfrgrg** [comment]
No tokens, no blockchain

**Comment by iduartgomez** [comment]
This software is not a blockchain neither has tokens. What's the point of this discussion?

**Reply by sfsfrgrg** [reply]
Meaning: reject blockchain, reject token

**Comment by iduartgomez** [comment]
And yes, it's free :)

**Comment by ple1n** [comment]
I think Locutus would be a great competitor/complement to blockchains. I hate the definition of web3=blockchain. 

1. Blockchain is centralized. There's no real DeFi when whales exist. In blockchain only miners maintain the consensus, while in Locutus all peers maintain the eventual consensus. It is not that equal or free.
2. Blockchain is costly. Gas fees, etc. 
3. Most p2p applications don't need a blockchain, but now everything seem to be on chain.  A decentralized GitHub or reddit won't need a blockchain. Because  consistency doesn't matter and Locutus has eventual consistency, which is enough in most cases. It's better to use a full-p2p solution in the first place rather than invent a L2 layer when the chain gets too congested.
4. Blockchain isn't partition-tolerant or really censorship-resistant. For example, it won't work in a mesh network, but Locutus would. Buying crypto is illegal in some countries.
5. Eventually some people have to pay for the servers. Either the users run servers by themselves *for themselves*, or others monetize through other ways, like ads, or otherwise the project is a Ponzi scheme.

In my opinion, Locutus will be an infrastructure like IPFS in the future. Lightweight applications will use it directly and others will use Locutus with IPFS (e.g. git), and blockchains will be used when necessary. The topology of it will look like fediverse since users tend to not serve things.

It is similar to Matrix, but more generalized. Anyway, instant messaging, forums, social media, etc. are just the same thing in different forms. DAO can also be established on Locutus, just like Matrix room moderation. 

Ideally Locutus would be the default platform for fediverse/full-p2p apps. IPFS is the decentralized filesystem, Ethereum is the strictly consistent world computer, and Locutus is the eventual-consistent world computer. Only computation/storage/bandwidth intensive tasks are sent to blockchain-based markets, like machine learning, search engine. The user or a relay node would connect to a wallet and start a deal. For usual tasks market doesn't make sense.

**Reply by sanity** [reply]
Mostly agree, I've given a lot of thought to how to implement a scalable and fast token/cryptocurrency on top of Locutus - I think it can be done if there is some mechanism for establishing that a contract had a particular state at a particular time, ie. the "arbiter" mechanism briefly mentioned in the README.md.

One weird phenomenon is the strong negative reaction I've seen whenever mentioning the possibility, people equate crypto with ponzi schemes and scams. This perhaps isn't surprising given how much of the cryptospace is ponzi schemes and scams, but it left me with the impression that we shouldn't place too much emphasis on this at least until we have a proof-of-concept implementation.

**Reply by delbonis** [reply]
I agree with the sentiment but I'd like to address a few points here:

>it won't work in a mesh network

Blockchains don't work if you're trying to be an active participant in consensus like as a miner since that does require good connectivity to other miners, but if you're a regular user you generally don't need strong network connections.

>Eventually some people have to pay for the servers.

Ideally the node software is lightweight enough that you can just embed it into the high-level software users normally run to interact with a protocol, so the sensible default has everyone do the good things and keeping the network decentralized.   Shitty "next-gen" blockchains make this impossible because it's just too expensive to run a node.

I like the IPFS comparison.
